source/slang/slang-emit.cpp

```diff
@@ -1585,6 +1645,9 @@ SlangResult CodeGenContext::emitEntryPointsSourceFromIR(ComPtr<IArtifact>& outAr
             linkedIR));
         
         auto irModule = linkedIR.module;
+
+        // TODO: How to store this in the artifact?
+        this->getProgram()->linkedIRModule = irModule;
```

Comment:
This was something I didn't figure out, please advise.

Reply:
I don't think we should return the linked IR. Instead, we should collect a list of mangled names that we have exported in the dxil, and attach that string list as the metadata of the output IR in linkAndOptimizeIR.

Reply:
Please checkout `collectMetadata` and store the list of mangled name into `ArtifactPostEmitMetadata`.

Reply:
Done!